101-azurefirewall-sandbox/azuredeploy.json

```diff
@@ -0,0 +1,422 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "virtualNetworkName": {
+      "type": "string",
+      "metadata": {
+        "description": "virtual network name"
+      }
+    },
+    "vnetAddressPrefix": {
+      "type": "string",
+      "defaultValue": "10.0.0.0/16",
+      "metadata": {
+        "description": "virtual network address range"
+      }
+    },
+    "serversSubnetPrefix": {
+      "type": "string",
+      "defaultValue": "10.0.2.0/24",
+      "metadata": {
+        "description": "Subnet prefix"
+      }
+    },
+    "azureFirewallSubnetPrefix": {
+      "type": "string",
+      "defaultValue": "10.0.1.0/24",
+      "metadata": {
+        "description": "AzureFirewallSubnet prefix"
+      }
+    },
+    "jumpboxSubnetPrefix": {
+      "type": "string",
+      "defaultValue": "10.0.0.0/24",
+      "metadata": {
+        "description": "Subnet prefix"
+      }
+    },
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "Username for the Virtual Machine."
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Password for the Virtual Machine."
+      }
+    },
+    "firewallName": {
+      "type": "string",
+      "defaultValue": "firewall1",
+      "metadata": {
+        "description": "azure firewall name"
+      }
+    },
+    "publicIPName": {
+      "type": "string",
+      "defaultValue": "publicIP1",
+      "metadata": {
+        "description": "Public IP address name"
+      }
+    },
+    "location": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]",
+      "metadata": {
+        "description": "Location for all resources."
+      }
+    }
+  },
+  "variables": {
+    "azureFirewallSubnetName": "AzureFirewallSubnet",
+    "jumpBoxSubnetName": "JumpboxSubnet",
+    "serversSubnetName": "ServersSubnet",
+    "jumpBoxPublicIPAddressName": "JumpHostPublicIP",
+    "jumpBoxNsgName": "JumpHostNSG",
+    "jumpBoxNicName": "JumpHostNic",
+    "jumpBoxSubnetId": "[resourceId('Microsoft.Network/virtualNetworks/subnets', parameters('virtualNetworkName'), variables('jumpBoxSubnetName'))]",
+    "serverNicName": "ServerNic",
+    "serverSubnetId": "[resourceId('Microsoft.Network/virtualNetworks/subnets', parameters('virtualNetworkName'), variables('serversSubnetName'))]",
+    "storageAccountName": "[concat(uniquestring(resourceGroup().id), 'sajumpbox')]",
+    "azfwRouteTableName": "AzfwRouteTable",
+    "nextHopIPPrefixStrLength": "[lastIndexOf(parameters('azureFirewallSubnetPrefix'), '.')]",
+    "nextHopIP": "[concat(substring(parameters('azureFirewallSubnetPrefix'), 0, variables('nextHopIPPrefixStrLength')), '.', '4')]",
```

Comment:
what happens with a different subnet range?  Will this be the correct math?

Reply:
This handles only the /24 subnet case and I have captured this in the comment section of route table where the nexthop is used.

Reply:
ok, then we need to either move the address ranges to variables or fix the math for the next hop so deployment doesn't fail because the user entered a different set of prefixes.